pkg/authz/authz.go

```diff
@@ -1,9 +1,68 @@
 package authz
 
 import (
+	"context"
+	"fmt"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
 	"github.com/openfga/openfga/pkg/logger"
 )
 
+type ctxKey string
+
+const (
+	// SkipAuthz is the key to store whether to skip authz check in the context.
+	skipAuthz ctxKey = "skip-authz-key"
+
+	// API methods.
+	ReadAuthorizationModel  = "ReadAuthorizationModel"
+	ReadAuthorizationModels = "ReadAuthorizationModels"
+	Read                    = "Read"
+	Write                   = "Write"
+	ListObjects             = "ListObjects"
+	StreamedListObjects     = "StreamedListObjects"
+	Check                   = "Check"
+	ListUsers               = "ListUsers"
+	WriteAssertions         = "WriteAssertions"
+	ReadAssertions          = "ReadAssertions"
+	WriteAuthorizationModel = "WriteAuthorizationModel"
+	ListStores              = "ListStores"
+	CreateStore             = "CreateStore"
+	GetStore                = "GetStore"
+	DeleteStore             = "DeleteStore"
+	Expand                  = "Expand"
+	ReadChanges             = "ReadChanges"
+
+	// Relations.
+	CanCallReadAuthorizationModels  = "can_call_read_authorization_models"
+	CanCallRead                     = "can_call_read"
+	CanCallWrite                    = "can_call_write"
+	CanCallListObjects              = "can_call_list_objects"
+	CanCallCheck                    = "can_call_check"
+	CanCallListUsers                = "can_call_list_users"
+	CanCallWriteAssertions          = "can_call_write_assertions"
+	CanCallReadAssertions           = "can_call_read_assertions"
+	CanCallWriteAuthorizationModels = "can_call_write_authorization_models"
+	CanCallListStores               = "can_call_list_stores"
+	CanCallCreateStore              = "can_call_create_stores"
+	CanCallGetStore                 = "can_call_get_store"
+	CanCallDeleteStore              = "can_call_delete_store"
+	CanCallExpand                   = "can_call_expand"
+	CanCallReadChanges              = "can_call_read_changes"
+
+	StoreType       = "store"
+	ApplicationType = "application"
+	SystemType      = "system"
+	RootSystemID    = "fga"
+)
+
+var (
+	ErrorResponse = &openfgav1.ForbiddenResponse{Code: 403, Message: "the principal is not authorized to perform the action"}
```

Comment:
**Question:** Is "principal" a newly introduced concept? Are there docs that use this term? If not, I'd like to converge on another existing term. Not sure what that would be though, user?

Reply:
@rhamzeh any thoughts?

Reply:
I don't have anything of the top of my head - I would keep principle for now and merge it and we can discuss this again before merging the final PR

Reply:
"principal" is the industry standard term for the authenticated entity that issued a request.